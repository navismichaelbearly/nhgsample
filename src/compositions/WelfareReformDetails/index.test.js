import React from 'react';
import { shallow } from 'enzyme';
import { WelfareReformDetailsComposition } from './';

jest.mock('date-fns/start_of_today', () => () => new Date('2018-10-01'));

describe('<WelfareReformDetailsComposition />', () => {
  let preventDefault;
  let props;
  let el;

  beforeEach(() => {
    preventDefault = jest.fn();
    props = {
      labels: {
        backButton: 'Back',
        saveButton: 'Save',
      },
      onBack: jest.fn(),
      onSubmit: jest.fn(),
      updatePageHeader: jest.fn(),
      getWelfareReformDetails: jest.fn(),
      getOptionSets: jest.fn(),
    };

    el = shallow(<WelfareReformDetailsComposition {...props} />);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render the page correctly', () => {
    expect(el).toMatchSnapshot();
  });

  it('should render the page correctly while loading', () => {
    el.setProps({ loading: true });
    expect(el).toMatchSnapshot();
  });

  it('should update the header', () => {
    expect(props.updatePageHeader).toHaveBeenCalled();
  });

  it('should set the default state', () => {
    expect(el.state()).toEqual({
      bedroomTax: false,
      benefitCap: false,
      claimStartDate: null,
      isDirty: {},
      paymentDay: null,
      universalCredit: false,
      apaStatus: null,
      ucArrearsPaymentStatus: null,
    });
  });

  it('should reset the form and call onBack when handleBackClick fires', () => {
    el.instance().handleBackClick();
    expect(props.onBack).toHaveBeenCalled();
  });

  describe('Form interaction', () => {
    beforeEach(() => {
      el.setState({
        universalCredit: false,
        claimStartDate: null,
        paymentDay: null,
        benefitCap: false,
        bedroomTax: false,
        apaStatus: null,
        ucArrearsPaymentStatus: null,
      });
    });

    describe('Form submit', () => {
      beforeEach(() => {
        el.setState({
          universalCredit: false,
          claimStartDate: null,
          paymentDay: null,
          benefitCap: false,
          bedroomTax: false,
          alternativePaymentArrangementStatusId: null,
          ucArrearsPaymentStatusId: null,
        });
      });

      it('should submit the form with the expected payload if all is well and it is an existing pause', () => {
        el.instance().handleFormSubmit({ preventDefault });
        expect(props.onSubmit).toHaveBeenCalledWith({
          universalCredit: false,
          claimStartDate: null,
          paymentDay: null,
          benefitCap: false,
          bedroomTax: false,
          alternativePaymentArrangementStatusId: null,
          ucArrearsPaymentStatusId: null,
        });
      });
    });
  });
});
