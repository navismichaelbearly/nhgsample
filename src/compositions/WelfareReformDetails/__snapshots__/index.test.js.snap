// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelfareReformDetailsComposition /> should render the page correctly 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <FormWrapper
      backButtonText="Back"
      disableBack={false}
      disablePropertyInformationEditMode={false}
      disableSubmit={false}
      formError={null}
      formName="welfareReformDetails"
      handleBackClick={[Function]}
      handleFormSubmit={[Function]}
      hidePropertyInformation={false}
      loading={false}
      otherActions={null}
      otherActionsLeft={null}
      submitButtonText="Save"
    >
      <div>
        <styled.h3>
          Welfare reform details
        </styled.h3>
        <styled.div
          data-bdd="universalCredit"
        >
          <styled.label>
            Universal Credit
          </styled.label>
          <x
            required={false}
          >
            <T
              checked={false}
              display="inline-block"
              error={false}
              hasError={false}
              onChange={[Function]}
            >
              Yes
            </T>
            <T
              checked={true}
              display="inline-block"
              error={false}
              hasError={false}
              onChange={[Function]}
            >
              No
            </T>
          </x>
        </styled.div>
        <styled.div
          data-bdd="benefitCap"
        >
          <styled.label>
            Benefit cap
          </styled.label>
          <x
            required={false}
          >
            <T
              checked={false}
              display="inline-block"
              error={false}
              hasError={false}
              onChange={[Function]}
            >
              Yes
            </T>
            <T
              checked={true}
              display="inline-block"
              error={false}
              hasError={false}
              onChange={[Function]}
            >
              No
            </T>
          </x>
        </styled.div>
        <styled.div
          data-bdd="bedroomTax"
        >
          <styled.label>
            Bedroom tax
          </styled.label>
          <x
            required={false}
          >
            <T
              checked={false}
              display="inline-block"
              error={false}
              hasError={false}
              onChange={[Function]}
            >
              Yes
            </T>
            <T
              checked={true}
              display="inline-block"
              error={false}
              hasError={false}
              onChange={[Function]}
            >
              No
            </T>
          </x>
        </styled.div>
      </div>
    </FormWrapper>
  </div>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<WelfareReformDetailsComposition /> should render the page correctly while loading 1`] = `
<withRouter(Connect(PageContentComposition))>
  <p
    isSmall={false}
  />
</withRouter(Connect(PageContentComposition))>
`;
